integrationtest/backend/lucene/src/test/java/org/hibernate/search/integrationtest/backend/lucene/fieldtype/LuceneFieldContentIT.java

```diff
@@ -0,0 +1,195 @@
+				)
+				.predicate( f -> f.matchAll() )
+				.build();
+
+		List<Document> result = query.execute().getHits();
+		Assertions.assertThat( result )
+				.hasSize( 2 )
+				.satisfies( containsDocument(
+						"ID:1",
+						doc -> doc.hasField( "string", "keyword" )
+								.hasField( "text", TEXT_1 )
+								.hasField( "integer", 739 )
+								.hasField( "longNumber", 739L )
+								.hasField( "bool", 1 )
+								.hasField( "nativeBool", 1 )
+								.andOnlyInternalFields()
+				) )
+				.satisfies( containsDocument(
+						"ID:2",
+						doc -> doc.hasField( "string", "anotherKeyword" )
+								.hasField( "text", TEXT_2 )
+								.hasField( "integer", 123 )
+								.hasField( "longNumber", 123L )
+								.hasField( "bool", 0 )
+								.hasField( "nativeBool", 0 )
+								.andOnlyInternalFields()
+				) );
+	}
+
+	private void initData() {
+		IndexWorkPlan<? extends DocumentElement> workPlan = indexManager.createWorkPlan();
+
+		workPlan.add( referenceProvider( "ID:1" ), document -> {
+			indexAccessors.string.write( document, "keyword" );
+			indexAccessors.text.write( document, TEXT_1 );
+			indexAccessors.integer.write( document, 739 );
+			indexAccessors.longNumber.write( document, 739L );
+			indexAccessors.bool.write( document, true );
+			indexAccessors.nativeBool.write( document, true );
+		} );
+		workPlan.add( referenceProvider( "ID:2" ), document -> {
+			indexAccessors.string.write( document, "anotherKeyword" );
+			indexAccessors.text.write( document, TEXT_2 );
+			indexAccessors.integer.write( document, 123 );
+			indexAccessors.longNumber.write( document, 123L );
+			indexAccessors.bool.write( document, false );
+			indexAccessors.nativeBool.write( document, false );
+		} );
+
+		workPlan.execute().join();
+	}
+
+	private static class IndexAccessors {
+
+		final IndexFieldAccessor<String> string;
+		final IndexFieldAccessor<String> text;
+		final IndexFieldAccessor<Integer> integer;
+		final IndexFieldAccessor<Long> longNumber;
+
+		final IndexFieldAccessor<Boolean> bool;
+		final IndexFieldAccessor<Boolean> nativeBool;
+
+		IndexAccessors(IndexSchemaElement root) {
+			string = root.field( "string", f -> f.asString().projectable( Projectable.YES ) ).createAccessor();
+			text = root.field( "text", f -> f.asString().analyzer( DefaultAnalysisDefinitions.ANALYZER_STANDARD.name ).projectable( Projectable.YES ) ).createAccessor();
+			integer = root.field( "integer", f -> f.asInteger().projectable( Projectable.YES ) ).createAccessor();
+			longNumber = root.field( "longNumber", f -> f.asLong().projectable( Projectable.YES ) ).createAccessor();
+			bool = root.field( "bool", f -> f.asBoolean().projectable( Projectable.YES ) ).createAccessor();
+
+			// Emulates what we do in LuceneBooleanFieldCodec:
+			// the external form is the Boolean,
+			// BUT we threat it as **Integer** for comparison operation (range, sort)
+			// and we store it as **Integer** as well.
+			nativeBool = root.field( "nativeBool", f ->
```

Comment:
Not sure why you decided to test this? You're basically testing code that you defined in your tests, so I doubt it is useful.

Reply:
Your're right, I'll remove it